Sources/SwiftProtobuf/Message+JSONAdditions.swift

```diff
@@ -34,6 +34,26 @@ public extension Message {
     return visitor.stringResult
   }
 
+  /// Returns a Data containing the UTF-8 JSON serialization of the message.
+  ///
+  /// Unlike binary encoding, presence of required fields is not enforced when
+  /// serializing to JSON.
+  ///
+  /// - Returns: A Data containing the JSON serialization of the message.
+  /// - Throws: `JSONEncodingError` if encoding fails.
+  func jsonUTF8Data() throws -> Data {
+    if let m = self as? _CustomJSONCodable {
+      let string = try m.encodedJSONString()
+      let data = string.data(using: String.Encoding.utf8)! // Cannot fail!
```

Comment:
In the future, it would be nice to get rid of calls to `string.data(using:)` because they're also pretty slow. I have some thoughts on how to do this for encoding that are related to the stream-based stuff I'd like to do for decoders in #420, but since it involves the use of some specific `String` APIs, I'm inclined to wait until Swift 4 is further along in development because those APIs are going to change a great deal and anything I do now might be wasted effort.